diff --git a/setup.py b/setup.py
index 14f93ca..693f2ef 100644
--- a/setup.py
+++ b/setup.py
@@ -18,6 +18,7 @@ import platform
 import subprocess
 import tempfile
 import shutil
+import sysconfig
 
 from setuptools import (
     setup,
@@ -79,10 +80,13 @@ class OTIO_build_ext(setuptools.command.build_ext.build_ext):
         if not os.path.exists(self.build_temp_dir):
             os.makedirs(self.build_temp_dir)
 
-        debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
-        self.build_config = ('Debug' if debug else 'Release')
+        self.build_config = os.environ.get("OTIO_CXX_BUILD_TYPE")
+        if not self.build_config:
+            debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
+            self.build_config = ('Debug' if debug else 'Release')
 
-        self.cmake_preflight_check()
+        # self.cmake_preflight_check()
+        self.announce(os.listdir(os.path.join(os.path.dirname(sys.executable), '..')), level=2)
         self.cmake_generate()
         self.cmake_install()
 
@@ -115,6 +119,9 @@ class OTIO_build_ext(setuptools.command.build_ext.build_ext):
             '-DBUILD_TESTING:BOOL=OFF',
             # Python modules wil be installed by setuptools.
             '-DOTIO_INSTALL_PYTHON_MODULES:BOOL=OFF',
+            # Not too sure why, but I have to pass that. This is pretty recent.
+            # Before it was working without issues.
+            f'-DPython_INCLUDE_DIR={sysconfig.get_path("include")}',
         ]
         if self.is_windows():
             if self.is_mingw():
@@ -264,13 +271,13 @@ class OTIO_build_py(setuptools.command.build_py.build_py):
     def run(self):
         super().run()
 
-        if not self.dry_run and not self.editable_mode:
-            # Only run when not in dry-mode (a dry run should not have any side effect)
-            # and in non-editable mode. We don't want to edit files when in editable
-            # mode because that could lead to modifications to the source files.
-            # Note that setuptools will set self.editable_mode to True
-            # when "pip install -e ." is run.
-            _append_version_info_to_init_scripts(self.build_lib)
+        # if not self.dry_run and not self.editable_mode:
+        #     # Only run when not in dry-mode (a dry run should not have any side effect)
+        #     # and in non-editable mode. We don't want to edit files when in editable
+        #     # mode because that could lead to modifications to the source files.
+        #     # Note that setuptools will set self.editable_mode to True
+        #     # when "pip install -e ." is run.
+        #     _append_version_info_to_init_scripts(self.build_lib)
 
 
 # copied from first paragraph of README.md
diff --git a/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt b/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
index d37da36..5559ce3 100644
--- a/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
+++ b/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
@@ -34,6 +34,8 @@ target_compile_options(_otio PRIVATE
      -Wno-range-loop-analysis>
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
      -Wno-unused-value>
+     $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
+     -sDISABLE_EXCEPTION_CATCHING=0>
      $<$<CXX_COMPILER_ID:MSVC>: /EHsc>
 )
 
