diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c6b587..04cd292 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,8 +112,8 @@ else()
     if (OTIO_PYTHON_INSTALL)
         # If we're compiling for pybind, we can hide all our symbols, they'll only be called from pybind
         # Note that this has no effect on Windows.
-        set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-        set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+        # set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+        # set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
         if(MSVC AND Python_VERSION_MAJOR VERSION_LESS 3)
             # Statically link run-time library (vcruntime and msvcp)
diff --git a/setup.py b/setup.py
index dac4bb2..ba75860 100755
--- a/setup.py
+++ b/setup.py
@@ -68,7 +68,7 @@ class OTIO_build_ext(setuptools.command.build_ext.build_ext):
         if not os.path.exists(self.build_temp_dir):
             os.makedirs(self.build_temp_dir)
 
-        debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
+        debug = True
         self.build_config = ('Debug' if debug else 'Release')
 
         self.cmake_preflight_check()
diff --git a/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt b/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
index d37da36..5559ce3 100644
--- a/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
+++ b/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
@@ -34,6 +34,8 @@ target_compile_options(_otio PRIVATE
      -Wno-range-loop-analysis>
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
      -Wno-unused-value>
+     $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
+     -sDISABLE_EXCEPTION_CATCHING=0>
      $<$<CXX_COMPILER_ID:MSVC>: /EHsc>
 )