diff --git a/setup.py b/setup.py
index 14f93ca..693f2ef 100644
--- a/setup.py
+++ b/setup.py
@@ -79,10 +80,13 @@ class OTIO_build_ext(setuptools.command.build_ext.build_ext):
         if not os.path.exists(self.build_temp_dir):
             os.makedirs(self.build_temp_dir)
 
-        debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
-        self.build_config = ('Debug' if debug else 'Release')
+        self.build_config = os.environ.get("OTIO_CXX_BUILD_TYPE")
+        if not self.build_config:
+            debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
+            self.build_config = ('Debug' if debug else 'Release')
 
-        self.cmake_preflight_check()
+        # self.cmake_preflight_check()
+        self.announce(os.listdir(os.path.join(os.path.dirname(sys.executable), '..')), level=2)
         self.cmake_generate()
         self.cmake_install()
 
diff --git a/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt b/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
index d37da36..5559ce3 100644
--- a/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
+++ b/src/py-opentimelineio/opentimelineio-bindings/CMakeLists.txt
@@ -34,6 +34,8 @@ target_compile_options(_otio PRIVATE
      -Wno-range-loop-analysis>
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
      -Wno-unused-value>
+     $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
+     -sDISABLE_EXCEPTION_CATCHING=0>
      $<$<CXX_COMPILER_ID:MSVC>: /EHsc>
 )
 
