diff --git a/CMakeLists.txt b/CMakeLists.txt
index fa74391..97c1c4e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,12 +114,12 @@ if(OTIO_SHARED_LIBS)
 else()
     message(STATUS "Building static libs")
     set(OTIO_SHARED_OR_STATIC_LIB "STATIC")
-    if (OTIO_PYTHON_INSTALL)
-        # If we're compiling for pybind, we can hide all our symbols, they'll only be called from pybind
-        # Note that this has no effect on Windows.
-        set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-        set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
-    endif()
+    # if (OTIO_PYTHON_INSTALL)
+    #     # If we're compiling for pybind, we can hide all our symbols, they'll only be called from pybind
+    #     # Note that this has no effect on Windows.
+    #     set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+    #     set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+    # endif()
 endif()
 
 set(OTIO_RESOLVED_CXX_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}")
